src/ios/removeFileFromProject.js

```diff
@@ -13,7 +14,13 @@ const removeFromProjectReferences = require('./removeFromProjectReferences');
  */
 module.exports = function removeFileFromProject(project, filePath) {
   const file = project.removeFromPbxFileReferenceSection(new PbxFile(filePath));
+  const projectRef = removeFromProjectReferences(project, file);
+
+  if (projectRef) {
+    removeProductGroup(project, projectRef.ProductGroup);
```

Comment:
So @Kureev for each projectReference there's a productGroup created. Added it here, so I can just remove by ProductGroup uuid :)